include/utils/utility.h

```diff
@@ -148,6 +155,45 @@ map_find(const Map & map,
   return it->second;
 }
 
+
+/**
+ * A replacement for std::vector::at(i) which is meant to be used with
+ * a macro, and, unlike at(), gives a proper line number and useful
+ * error message when the index is past the end.
+ */
+template<typename Vector>
+inline
+typename Vector::reference &
+vector_at(Vector & vec,
+          typename Vector::size_type i,
+          const char * filename,
+          int line_number)
+{
+  if (i >= vec.size())
+    libmesh_error_msg("vec_at() error: Index " + std::to_string(i) +
+                      " past end of vector in file " + std::string(filename) +
+                      " on line " + std::to_string(line_number));
+  return vec[i];
+}
+
+/**
+ * Same as above, but for const inputs.
+ */
+template<typename Vector>
+inline
+typename Vector::const_reference &
+vector_at(const Vector & vec,
+          typename Vector::size_type i,
+          const char * filename,
+          int line_number)
+{
+  if (i >= vec.size())
+    libmesh_error_msg("vec_at() error: Index " + std::to_string(i) +
```

Comment:
I'd just using streaming `<<`, cut out all the `std::to_string` which shouldn't be necessary.

Reply:
Good point.  Will do.